### PR TITLE
Issue136

### PR DESCRIPTION
**Pull request by xmclark**
Add a unit test for the `New::execute` file system calls. We use the [assert_fs](https://docs.rs/assert_fs/0.2.1/assert_fs/index.html) library to produce an ephemeral file system. It works pretty well.

### CODE REVIEW

File: hatch/src/cli/commands/new.rs [hunk 1]
```diff
@@ -8,13 +8,26 @@ use project::{CompilerOptions, Project};
...
+use std::path::Path;
 
 pub struct New;
 
 impl<'new> New {
     pub fn new() -> New {
         New
     }
+
+    pub fn write_directory_structure(&self, dir_path: &Path) -> HatchResult<()> {
```
Comment (DannyPeck) [inline hatch/src/cli/commands/new.rs:20]: Nice!

File: hatch/src/cli/commands/new.rs [hunk 2]
```diff
@@ -53,3 +61,30 @@ impl<'command> Command<'command> for New {
...
+        use super::New;
+        use assert_fs::TempDir;
+        use assert_fs::prelude::*;
+        use predicates::prelude::*;
+        let temp = TempDir::new().unwrap();
+        let dir_path = temp.path().to_owned();
+        let new = New::new();
+        let result = new.write_directory_structure(&dir_path);
+        let is_dir = predicate::path::is_dir();
+        temp.child("src").assert(&is_dir);
```
Comment (DannyPeck) [inline hatch/src/cli/commands/new.rs:79]: This is really neat!  So is "is_dir" a function? And assert can take a function?

**Review by DannyPeck — APPROVED** [review]

File: hatch/src/cli/commands/new.rs (hunk 2 above)
Comment (xmclark) [reply hatch/src/cli/commands/new.rs:79]: Correct.

Comment (xmclark) [reply hatch/src/cli/commands/new.rs:79]: the predicates library might be useful for other things too, it is not specific to assert_fs

Comment (DannyPeck) [reply hatch/src/cli/commands/new.rs:79]: That's awesome and definitely!

**Pull request merged**